lib/discourse_chat/provider/slack/slack_message.rb

```diff
@@ -29,11 +29,27 @@ def url
     def text
       text = @raw['text'].nil? ? "" : @raw['text']
 
+      pre = {}
+
+      # Extract code blocks and replace with placeholder
+      text = text.gsub(/```(.*)```/m) do |match|
```

Comment:
You might want to be lazy, just in case there are more than one code block? (is that even possible?)

```suggestion
      text = text.gsub(/```(.*?)```/m) do |match|
```